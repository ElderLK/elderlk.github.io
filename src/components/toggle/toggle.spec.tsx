import React from 'react'
import { render, screen, act } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Toggle from './toggle.component'

describe('<Toogle />', () => {
  test('Should match snapshot', () => {
    const { container } = render(<Toggle label="my-tootle" />)

    expect(container.firstChild).toMatchSnapshot()
  })

  test('Should render correctly', () => {
    render(<Toggle label="my-tootle" />)

    expect(
      screen.getByRole('label', { name: /my-tootle/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('checkbox')).toBeInTheDocument()
  })

  test('Should render the component as controlled and switch value', async () => {
    const onToggle = jest.fn()

    render(<Toggle label="my-tootle" checked={false} onToggle={onToggle} />)

    expect(screen.getByRole('checkbox')).not.toBeChecked()

    await act(async () => {
      await userEvent.click(screen.getByRole('checkbox'))
    })

    expect(screen.getByRole('checkbox')).toBeChecked()
  })

  test('Should render the component as uncontrolled and switch value', async () => {
    const ref = React.createRef<HTMLInputElement>()

    render(<Toggle label="my-tootle" ref={ref} />)

    expect(screen.getByRole('checkbox')).not.toBeChecked()
    expect(ref.current?.checked).toBe(false)

    await act(async () => {
      await userEvent.click(screen.getByRole('checkbox'))
    })

    expect(screen.getByRole('checkbox')).toBeChecked()
    expect(ref.current?.checked).toBe(true)
  })
})
