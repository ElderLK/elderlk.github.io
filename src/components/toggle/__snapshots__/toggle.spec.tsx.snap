// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toogle /> Should match snapshot 1`] = `
<label
  class="relative inline-flex cursor-pointer items-center"
>
  <input
    class="peer sr-only"
    type="checkbox"
    value=""
  />
  <div
    class="peer relative rounded-full bg-gray-300 peer-checked:bg-blue-600 peer-focus:outline-none peer-focus:ring-4 peer-focus:ring-blue-300 dark:border-gray-600 dark:bg-gray-700 dark:peer-focus:ring-blue-800 h-6 w-11"
  >
    <span
      class="absolute flex items-center rounded-full border border-gray-300 bg-white text-gray-950 transition-all left-[2px] top-[2px] h-5 w-5"
    />
  </div>
  <span
    class="ml-3 text-sm font-medium text-gray-900 dark:text-gray-300"
    role="label"
  >
    my-tootle
  </span>
</label>
`;
